### DISCUSSION TITLE
Maintenance system

### DISCUSSION THREAD

**Discussion by LorickLaBrique** [pelican-dev/panel#214] [Ideas]
I don’t know if it’s envisaged but an idea : make a maintenance system to replace the php artisan down

In the settings page, activate a button that turn the panel in a maintenance state to redirect all users who wish to access the panel to a maintenance page. But the administrator must have access to the panel and I don’t know how to do this

**Comment by Boy132** [comment]
FYI, there already is a way to [bypass the maintenance mode](https://laravel.com/docs/11.x/configuration#bypassing-maintenance-mode) by providing a secret, e.g. `php artisan down --secret="youshallnotpass"`. If you want to generate a random secret you can use `php artisan down --with-secret` instead.
Anyone who knows the secret, e.g. admins, can then enter the panel via `https://panel.example.com/youshallnotpass`

**Reply by LorickLaBrique** [reply]
![image](https://github.com/pelican-dev/panel/assets/96184599/d94a5982-c6c4-4db8-b8b0-ddd4b0549f55)

@Boy132

**Comment by LorickLaBrique** [comment]
Yes but make a button in the settings panel can be a cool thing

**Comment by Boy132** [comment]
> ![image](https://private-user-images.githubusercontent.com/96184599/330102681-d94a5982-c6c4-4db8-b8b0-ddd4b0549f55.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2MTM0ODgsIm5iZiI6MTcxNTYxMzE4OCwicGF0aCI6Ii85NjE4NDU5OS8zMzAxMDI2ODEtZDk0YTU5ODItYzZjNC00ZGI4LWI4YjAtZGRkNGIwNTQ5ZjU1LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxM1QxNTEzMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MWY3NGQ0MGRiZGJlOTE2YTYxZTIxMmUzMGQ1OGRlNDc4NGYyYmNjOGUyY2FjYWE1Yzc5Y2Y4OWZlMmYxMzA0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GfBodlUBDmTQR3JuUx5re2Bt0rB2angVs1DbKXmFpMg)
> 
> @Boy132

You are missing php extensions or are using the wrong php version.

**Comment by LorickLaBrique** [comment]
Yes I had PHP 8.3 installed While I have PHP 8.1

**Comment by LorickLaBrique** [comment]
@lancepioch What do you think?

**Comment by rmartinoscar** [comment]
Isn't the goal of a global Maintenance to shut down the panel ?
If you want to keep access to it just enable Maintenance mode in the Edit Node page.
If you want to edit the panel, do it outside of prod, then when its production ready `php artisan down` swap folders and `php artisan up`, and if you really want to access it when its down follow @Boy132's reply although not recommended

**Comment by notAreYouScared** [comment]
We suggest using `php artisan down` and `php artisan up` along with what @Boy132 said below

> e.g. `php artisan down --secret="youshallnotpass"`. If you want to generate a random secret you can use `php artisan down --with-secret` instead. Anyone who knows the secret, e.g. admins, can then enter the panel via `https://panel.example.com/youshallnotpass`
